hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKey.java

```diff
@@ -86,6 +86,21 @@ default long getNonce() {
    */
   long getOrigLogSeqNum();
 
+    /**
+     * Return a named String value injected into the WALKey during processing, such as by a
+     * coprocessor
+     * @param attributeKey The key of a key / value pair
+     */
+  default byte[] getExtendedAttribute(String attributeKey){
+    return null;
+  }
+
+    /**
+     * Returns a map of all extended attributes injected into this WAL key.
+     */
+  default Map<String, byte[]> getExtendedAttributes() {
```

Comment:
Missing setters?

Reply:
@apurtell - WALKey's interface comments specifically say that setters aren't permitted. They're meant to be immutable, so the extended attributes will be set during construction of the WALKeyImpl